src/service-override/views.ts

```diff
@@ -304,17 +340,39 @@ function registerEditorPane (options: EditorPanelOption): { disposable: IDisposa
     }
   }
 
-  const disposable = Registry.as<IEditorPaneRegistry>(EditorExtensions.EditorPane).registerEditorPane(
+  const disposableStore = new DisposableStore()
+
+  disposableStore.add(Registry.as<IEditorPaneRegistry>(EditorExtensions.EditorPane).registerEditorPane(
     EditorPaneDescriptor.create(
       CustomEditorPane,
       options.id,
       options.name
     ),
-    [new SyncDescriptor(CustomEditorInput)])
+    [new SyncDescriptor(CustomEditorInput)]))
 
   return {
-    disposable,
-    CustomEditorInput
+    disposable: disposableStore,
+    CustomEditorInput,
+    async registerEditor (globPattern: string, registerEditorOptions: RegisterEditorOptions = {}): Promise<IDisposable> {
+      const resolverService = await getService(IEditorResolverService)
+      return resolverService.registerEditor(
```

Comment:
shouldn't the returned disposable be disposed when the whole editor pane is disposed?